### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by fpsapc**
- Add the following validations:
- For the User model:
  - Name must not be blank.
  - PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
  - Title must not be blank.
  - Title must not exceed 250 characters.
  - CommentsCounter must be an integer greater than or equal to zero.
  - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -5,6 +5,10 @@ class Post < ApplicationRecord
 
   after_create :update_author_posts_counter
 
+  validates :title, presence: true
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
```
Comment (TadesseDev) [inline app/models/post.rb]: - [x] Your validation is already great @fpsapc, but I notice you miss the validation for `Title must not exceed 250 characters.`. this is in addition to the one you already have `Title must not be blank.`. Please make sure the title does not exceed 250 characters.

File: spec/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  describe 'Validations For the Like model' do
+    before(:each) do
+      @user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.',
```
Comment (TadesseDev) [inline spec/like_spec.rb:6]: - [ ] Dear @fpsapc, Please remember all custom methods in the models should have a corresponding unit test. And for the `Like` model we need to test the `update_post_likes_counter` method.

File: spec/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  describe 'Validations For the Comment model' do
+    before(:each) do
+      @user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.',
+                          posts_counter: 0)
```
Comment (TadesseDev) [inline spec/comment_spec.rb:7]: - [ ] Dear @fpsapc, Please remember all custom methods in the models should have a corresponding unit test. And for the `Comment` model we need to test the `update_post_comments_counter` method.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @fpsapc 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] Most of the requirements are in place ✅
 - [x] You have written a clean code ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @fpsapc 👋 

### Things to appreciate
- [ ] Implemented the things requested by the previous code reviewer 👍 
- [ ] Fulfilled project requirements 👍 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**